app/src/features/surveys/view/SurveyMap.tsx

```diff
@@ -1,29 +1,177 @@
+  label: string;
+  value: string;
+}
+
+export interface ISurveyMapSupplementaryLayer {
+  /**
+   * The name of the layer
+   */
+  layerName: string;
+  /**
+   * The array of map points
+   */
+  mapPoints: ISurveyMapPoint[];
+  /**
+   * The title of the feature type displayed in the popup
+   */
+  popupRecordTitle: string;
+}
+
+export interface ISurveyMapPoint {
+  /**
+   * Unique key for the point
+   */
+  key: string;
+  /**
+   * The geometric feature to display
+   */
+  feature: Feature;
+
+  /**
+   * Callback to fetch metadata, which is fired when the geometry's popup
+   * is opened
+   */
+  onLoadMetadata: () => Promise<ISurveyMapPointMetadata[]>;
+  /**
+   * Optional link that renders a button to view/manage/edit the data
+   * that the geometry belongs to
+   */
+  link?: string;
+}
 
 interface ISurveyMapProps {
-  mapPoints: INonEditableGeometries[];
+  supplementaryLayers: ISurveyMapSupplementaryLayer[];
   isLoading: boolean;
 }
 
-// TODO: need a way to pass in the map dimensions depending on the screen size
+const useStyles = makeStyles((theme: Theme) => ({
+  popup: {
+    '& a': {
+      color: theme.palette.primary.contrastText
+    },
+    '& p': {
+      margin: 0
+    }
+  }
+}));
+
+interface ISurveyMapPopupProps {
+  isLoading: boolean;
+  title: string;
+  metadata: ISurveyMapPointMetadata[];
+}
+
+const SurveyMapPopup = (props: ISurveyMapPopupProps) => {
+  return (
+    <Box>
+      {props.isLoading ? (
+        <Box position="absolute" top="0" left="0" right="0" sx={{ opacity: 1 }}>
+          <Typography
+            component="div"
+            variant="body2"
+            fontWeight={700}
+            sx={{
+              pr: 3,
+              whiteSpace: 'nowrap',
+              overflow: 'hidden',
+              textOverflow: 'ellipsis'
+            }}>
+            <Skeleton></Skeleton>
+          </Typography>
+          <Box mt={1} mb={0}>
+            <Stack flexDirection="row" alignItems="flex-start" gap={1} sx={{ typography: 'body2' }}>
+              <Skeleton width="80px" />
+              <Skeleton sx={{ flex: '1 1 auto' }} />
+            </Stack>
+            <Stack flexDirection="row" alignItems="flex-start" gap={1} sx={{ typography: 'body2' }}>
+              <Skeleton width="80px" />
+              <Skeleton sx={{ flex: '1 1 auto' }} />
+            </Stack>
+          </Box>
+        </Box>
+      ) : (
+        <Box>
+          <Typography
+            component="div"
+            variant="body2"
+            fontWeight={700}
+            sx={{
+              pr: 4,
+              whiteSpace: 'nowrap',
+              overflow: 'hidden',
+              textOverflow: 'ellipsis',
+              textTransform: 'uppercase'
+            }}>
+            {props.title}
+          </Typography>
+          <Box component="dl" mt={1} mb={0}>
+            {props.metadata.map((metadata, index) => (
+              <Stack
+                key={`${metadata.label}-${index}`}
```

Comment:
Is there a better key to use instead of the index?

Reply:
I think if we can get away from using the index it would be prefered. Just so we dont have to re-render the whole list if the order of the items change.